clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -51,9 +52,15 @@ static void get_version(const sycl::device &dev, int &major, int &minor) {
       break;
     i++;
   }
-  i++;
-  minor = std::stoi(&(ver[i]));
-}
+  if (i < ver.size()) {
```

Comment:
Similar PR as https://github.com/oneapi-src/SYCLomatic/pull/1782 , while #1782 adding more checkers in major. Will merge 1782. Thanks.

Reply:
Thanks, yes #1782 is more applicable. Closing